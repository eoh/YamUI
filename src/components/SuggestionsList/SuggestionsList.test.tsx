/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { default as SuggestionsList, SuggestionsListProps } from '.';
import { LayoutListItem } from '../LayoutList';

describe('<SuggestionsList />', () => {
  let rendered: ShallowWrapper<SuggestionsListProps>;
  let onItemSelected: jest.Mock;
  let options: Partial<SuggestionsListProps>;

  const getProps = (overrides: Partial<SuggestionsListProps>) => {
    onItemSelected = jest.fn();

    const defaultProps = {
      onItemSelected,
      searchText: 'searchText',
      loadingText: 'loadingText',
      noResultsText: 'noResultsText',
      isLoading: false,
      groupedItems: [],
      selectedId: 2,
    };

    return {
      ...defaultProps,
      ...overrides,
    };
  };

  beforeEach(() => {
    options = {};
  });

  describe('when isLoading=true', () => {
    beforeEach(() => {
      options = {
        ...options,
        isLoading: true,
      };
    });

    describe('without results', () => {
      beforeEach(() => {
        rendered = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('renders as expected', () => {
        expect(rendered).toMatchSnapshot();
      });
    });

    describe('with results', () => {
      beforeEach(() => {
        options = {
          ...options,
          groupedItems: [
            {
              title: 'groupTitle',
              items: [
                {
                  id: 'id',
                  imageUrl: 'imageUrl',
                  name: 'name',
                  description: 'description',
                },
              ],
            },
          ],
        };
        rendered = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('renders as expected', () => {
        expect(rendered).toMatchSnapshot();
      });

      describe('when onHover is called', () => {
        beforeEach(() => {
          rendered.find('SuggestionsListItem').simulate('hover', '1');
        });

        it('updates the state', () => {
          expect(rendered.state()).toMatchSnapshot();
        });
      });

      describe('with state.hoveredId=id', () => {
        beforeEach(() => {
          rendered.setState({ hoveredId: 'id' });
        });

        it('renders as expected', () => {
          expect(rendered).toMatchSnapshot();
        });

        describe('when the mouse leaves', () => {
          beforeEach(() => {
            rendered
              .find(LayoutListItem)
              .at(1)
              .simulate('mouseLeave');
          });

          it('renders as expected', () => {
            expect(rendered).toMatchSnapshot();
          });
        });
      });
    });
  });

  describe('when isLoading=false', () => {
    beforeEach(() => {
      options = {
        ...options,
        isLoading: false,
      };
    });

    describe('without results', () => {
      beforeEach(() => {
        rendered = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('renders as expected', () => {
        expect(rendered).toMatchSnapshot();
      });
    });

    describe('with results', () => {
      beforeEach(() => {
        options = {
          ...options,
          groupedItems: [
            {
              title: 'groupTitle',
              items: [
                {
                  id: 'id',
                  imageUrl: 'imageUrl',
                  name: 'name',
                  description: 'description',
                },
              ],
            },
          ],
        };
        rendered = shallow(<SuggestionsList {...getProps(options)} />);
      });

      it('renders as expected', () => {
        expect(rendered).toMatchSnapshot();
      });

      describe('when onHover is called', () => {
        beforeEach(() => {
          rendered.find('SuggestionsListItem').simulate('hover', '1');
        });

        it('updates the state', () => {
          expect(rendered.state()).toMatchSnapshot();
        });
      });

      describe('with state.hoveredId=id', () => {
        beforeEach(() => {
          rendered.setState({ hoveredId: 'id' });
        });

        it('renders as expected', () => {
          expect(rendered).toMatchSnapshot();
        });

        describe('when the mouse leaves', () => {
          beforeEach(() => {
            rendered
              .find(LayoutListItem)
              .at(1)
              .simulate('mouseLeave');
          });

          it('renders as expected', () => {
            expect(rendered).toMatchSnapshot();
          });
        });
      });
    });
  });
});
