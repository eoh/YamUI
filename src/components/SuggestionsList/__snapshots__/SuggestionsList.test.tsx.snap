// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestionsList /> when isLoading=false with results renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        padding-bottom: 4px;
        padding-top: 4px;
        width: 280px;
      }
>
  <LayoutList
    direction="vertical"
  >
    <LayoutListItem
      key="groupTitle"
    >
      <LayoutList
        direction="vertical"
      >
        <LayoutListItem
          key="id"
        >
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </LayoutListItem>
      </LayoutList>
    </LayoutListItem>
  </LayoutList>
</div>
`;

exports[`<SuggestionsList /> when isLoading=false with results when onHover is called updates the state 1`] = `null`;

exports[`<SuggestionsList /> when isLoading=false with results with state.hoveredId=id renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        padding-bottom: 4px;
        padding-top: 4px;
        width: 280px;
      }
>
  <LayoutList
    direction="vertical"
  >
    <LayoutListItem
      key="groupTitle"
    >
      <LayoutList
        direction="vertical"
      >
        <LayoutListItem
          key="id"
        >
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </LayoutListItem>
      </LayoutList>
    </LayoutListItem>
  </LayoutList>
</div>
`;

exports[`<SuggestionsList /> when isLoading=false with results with state.hoveredId=id when the mouse leaves renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        padding-bottom: 4px;
        padding-top: 4px;
        width: 280px;
      }
>
  <LayoutList
    direction="vertical"
  >
    <LayoutListItem
      key="groupTitle"
    >
      <LayoutList
        direction="vertical"
      >
        <LayoutListItem
          key="id"
        >
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </LayoutListItem>
      </LayoutList>
    </LayoutListItem>
  </LayoutList>
</div>
`;

exports[`<SuggestionsList /> when isLoading=false without results renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        width: 280px;
      }
>
  <Block
    padding="large"
    textAlign="center"
    textSize="small"
  >
    <Text
      color="metadata"
    >
      noResultsText
    </Text>
  </Block>
</div>
`;

exports[`<SuggestionsList /> when isLoading=true with results renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        padding-top: 4px;
        width: 280px;
      }
>
  <LayoutList
    direction="vertical"
  >
    <LayoutListItem
      key="groupTitle"
    >
      <LayoutList
        direction="vertical"
      >
        <LayoutListItem
          key="id"
        >
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </LayoutListItem>
      </LayoutList>
    </LayoutListItem>
  </LayoutList>
  <div
    className=

        {
          border-top: 1px solid #f3f5f8;
          padding-bottom: 14px;
          padding-top: 13px;
        }
  >
    <Spinner
      isCentered={true}
      text="loadingText"
    />
  </div>
</div>
`;

exports[`<SuggestionsList /> when isLoading=true with results when onHover is called updates the state 1`] = `null`;

exports[`<SuggestionsList /> when isLoading=true with results with state.hoveredId=id renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        padding-top: 4px;
        width: 280px;
      }
>
  <LayoutList
    direction="vertical"
  >
    <LayoutListItem
      key="groupTitle"
    >
      <LayoutList
        direction="vertical"
      >
        <LayoutListItem
          key="id"
        >
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </LayoutListItem>
      </LayoutList>
    </LayoutListItem>
  </LayoutList>
  <div
    className=

        {
          border-top: 1px solid #f3f5f8;
          padding-bottom: 14px;
          padding-top: 13px;
        }
  >
    <Spinner
      isCentered={true}
      text="loadingText"
    />
  </div>
</div>
`;

exports[`<SuggestionsList /> when isLoading=true with results with state.hoveredId=id when the mouse leaves renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        padding-top: 4px;
        width: 280px;
      }
>
  <LayoutList
    direction="vertical"
  >
    <LayoutListItem
      key="groupTitle"
    >
      <LayoutList
        direction="vertical"
      >
        <LayoutListItem
          key="id"
        >
          <SuggestionsListItem
            description="description"
            id="id"
            imageUrl="imageUrl"
            isSelected={false}
            name="name"
            onSelect={[MockFunction]}
            searchText="searchText"
          />
        </LayoutListItem>
      </LayoutList>
    </LayoutListItem>
  </LayoutList>
  <div
    className=

        {
          border-top: 1px solid #f3f5f8;
          padding-bottom: 14px;
          padding-top: 13px;
        }
  >
    <Spinner
      isCentered={true}
      text="loadingText"
    />
  </div>
</div>
`;

exports[`<SuggestionsList /> when isLoading=true without results renders as expected 1`] = `
<div
  className=
      y-suggestions-list
      y-hc-border
      {
        border-radius: 1px;
        border-width: 0px;
        border: none;
        box-shadow: 0 0 12px rgba(0, 0, 0, 0.2);
        width: 280px;
      }
>
  <div
    className=

        {
          padding-bottom: 14px;
          padding-top: 14px;
        }
  >
    <Spinner
      isCentered={true}
      text="loadingText"
    />
  </div>
</div>
`;
